[Bug][202405]: Failed: Orch CPU utilization  > orch cpu threshold 10 before link flap

### Issue Description

The `platform_tests/link_flap/test_cont_link_flap.py` test checks whether the CPU% utilization is less than 10% before it begins flapping the interfaces. However, this check fails intermittently on a modular chassis in T2 topology.

### Results you see

```
E       Failed: Orch CPU utilization  > orch cpu threshold 10 before link flap
--------------------------------------------------
	 platform_tests 1
--------------------------------------------------
	 2024-12-15 23:36:09: platform_tests/link_flap/test_cont_link_flap.py::TestContLinkFlap::test_cont_link_flap[cmp227-5]
```

### Results you expected to see

We expect it to pass.

### Is it platform specific

generic

### Relevant log output

_No response_

### Output of `show version`

```shell
SONiC Software Version: SONiC.branch.202405-ars.7e95d019-buildimage.origin.202405-nightly-dbg-2024.12.16.22.44
SONiC OS Version: 12
Distribution: Debian 12.8
Kernel: 6.1.0-22-2-amd64
Build commit: 5ac5577d1
Build date: Tue Dec 17 09:27:08 UTC 2024
Built by: jenkins@jenkins-arsonic-k8s-6-j8lwz

Platform: x86_64-arista_7800_sup
HwSKU: Arista-7804R3-FM
ASIC: broadcom
ASIC Count: 12
Serial Number: SSN20290119
Model Number: DCS-7800-SUP1A
Hardware Revision: 01.0a
Uptime: 18:56:06 up 2 days, 19:32,  1 user,  load average: 6.03, 6.00, 6.00
Date: Fri 20 Dec 2024 18:56:06
```


### Attach files (if any)

_No response_

--- a/tests/platform_tests/link_flap/test_cont_link_flap.py
+++ b/tests/platform_tests/link_flap/test_cont_link_flap.py
@@ -101,7 +101,12 @@ class TestContLinkFlap(object):
 
         # Make Sure Orch CPU < orch_cpu_threshold before starting test.
         logging.info("Make Sure orchagent CPU utilization is less that %d before link flap", orch_cpu_threshold)
-        pytest_assert(wait_until(600, 2, 0, check_orch_cpu_utilization, duthost, orch_cpu_threshold),
+        if 't2' in tbinfo['topo']['name']:
+            # To address bug 16186
+            wait_timeout = 600
+        else:
+            wait_timeout = 100
+        pytest_assert(wait_until(wait_timeout, 2, 0, check_orch_cpu_utilization, duthost, orch_cpu_threshold),
                       "Orch CPU utilization {} > orch cpu threshold {} before link flap"
                       .format(duthost.shell("show processes cpu | grep orchagent | awk '{print $9}'")["stdout"],
                               orch_cpu_threshold))

--- a/tests/platform_tests/link_flap/test_cont_link_flap.py
+++ b/tests/platform_tests/link_flap/test_cont_link_flap.py
@@ -101,7 +101,7 @@ class TestContLinkFlap(object):
 
         # Make Sure Orch CPU < orch_cpu_threshold before starting test.
         logging.info("Make Sure orchagent CPU utilization is less that %d before link flap", orch_cpu_threshold)
-        pytest_assert(wait_until(100, 2, 0, check_orch_cpu_utilization, duthost, orch_cpu_threshold),
+        pytest_assert(wait_until(600, 2, 0, check_orch_cpu_utilization, duthost, orch_cpu_threshold),
                       "Orch CPU utilization {} > orch cpu threshold {} before link flap"
                       .format(duthost.shell("show processes cpu | grep orchagent | awk '{print $9}'")["stdout"],
                               orch_cpu_threshold))

--- a/tests/platform_tests/link_flap/test_cont_link_flap.py
+++ b/tests/platform_tests/link_flap/test_cont_link_flap.py
@@ -102,7 +102,8 @@ class TestContLinkFlap(object):
         # Make Sure Orch CPU < orch_cpu_threshold before starting test.
         logging.info("Make Sure orchagent CPU utilization is less that %d before link flap", orch_cpu_threshold)
         if 't2' in tbinfo['topo']['name']:
-            # To address bug 16186
+            # In T2 topology, if the test is run on uplink LC first, it needs more time for the CPU to cool down
+            # More details in bug 16186
             wait_timeout = 600
         else:
             wait_timeout = 100

